ReferenceError: window is not defined when using worker_threads with NodeJS

# **Describe the bug**
I'm trying to run a very simple worker_threads example (code is below). While debugging I receive the following error. Because the error/exception is happening in the worker thread I cannot break on it without enabling `break on uncaught exceptions`.

Without break on uncaught exceptions this is the only ouput I recieve:
```
C:\src\worker
{ isMainThread: true }
Worker stopped with exit code 1
```
Breaking on uncaught exceptions I'm able to see these errors in the worker thread:
```
Exception has occurred: ReferenceError: window is not defined
  at eval (eval at checkNotElectron (c:\Users\<username>\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\ms-vscode.js-debug\src\bootloader.bundle.js:2:95083), <anonymous>:1:1)
    at checkNotElectron (bootloader.bundle.cdp:2:95083)
    at bootloader.bundle.cdp:2:96247
    at Array.some (<anonymous>)
    at checkAll (bootloader.bundle.cdp:2:96237)
    at bootloader.bundle.cdp:2:90808
    at 6679 (bootloader.bundle.cdp:2:93349)
    at __webpack_require__ (bootloader.bundle.cdp:2:99703)
    at bootloader.bundle.cdp:2:99856
    at Object.<anonymous> (bootloader.bundle.cdp:2:99883)
```

For the life of me I cannot figure out why the debugger is trying to check for window in this case or load any bundlers. I also notice _sometimes_ other errors are occurring. such as the below. Notice all of these are happening in a bundler.

```
Error: Cannot find module 'bufferutil'
    at bootloader.bundle.cdp:2:37524
    at 977 (bootloader.bundle.cdp:2:37603)
    at __webpack_require__ (bootloader.bundle.cdp:2:99703)
    at 5196 (bootloader.bundle.cdp:2:42800)
    at __webpack_require__ (bootloader.bundle.cdp:2:99703)
    at 8762 (bootloader.bundle.cdp:2:66776)
    at __webpack_require__ (bootloader.bundle.cdp:2:99703)
    at 8777 (bootloader.bundle.cdp:2:36628)
    at __webpack_require__ (bootloader.bundle.cdp:2:99703)
    at 7993 (bootloader.bundle.cdp:2:82774) {code: 'MODULE_NOT_FOUND', stack: 'Error: Cannot find module 'bufferutil'
    at…)
    at 7993 (bootloader.bundle.cdp:2:82774)', message: 'Cannot find module 'bufferutil''}
```
```
Error: Cannot find module 'undefined'
    at Function.t [as resolve] (bootloader.bundle.cdp:2:96505)
    at checkProcessFilter (bootloader.bundle.cdp:2:95324)
    at bootloader.bundle.cdp:2:96247
    at Array.some (<anonymous>)
    at checkAll (bootloader.bundle.cdp:2:96237)
    at bootloader.bundle.cdp:2:90808
    at 6679 (bootloader.bundle.cdp:2:93349)
    at __webpack_require__ (bootloader.bundle.cdp:2:99703)
    at bootloader.bundle.cdp:2:99856
    at Object.<anonymous> (bootloader.bundle.cdp:2:99883) {code: 'MODULE_NOT_FOUND', stack: 'Error: Cannot find module 'undefined'
    at …t.<anonymous> (bootloader.bundle.cdp:2:99883)', message: 'Cannot find module 'undefined''}
```

# **To Reproduce**
The following are `main.js`, `worker.js`, and and the `package.json` I'm using. Enable `break on uncaught exceptions` and run main.js using VSCode js-debugger. The debugger will sudden pause, check the call stack window, you will see a `Worker 1` thread, click continue on the thread to see the resulting errors from above

![image](https://user-images.githubusercontent.com/122838399/222932389-2a52b2cc-11aa-4c85-9cc8-7d0987e90049.png)


```javascript
import { Worker, isMainThread } from "node:worker_threads";

function runService(workerData) {

  console.log(process.cwd());
  const worker = new Worker('./worker.js', { workerData });
  
  worker.on("message", incoming => console.log({ incoming }));
  worker.on("error", code => new Error(`Worker error with exit code ${code}`));
  worker.on("exit", code =>
    console.log(`Worker stopped with exit code ${code}`)
  );
  worker.postMessage("once");
  worker.postMessage("twice");
  worker.postMessage("three times");
  worker.postMessage("exit");
  setTimeout(() => worker.postMessage("you won't see me"), 100);
}

async function run() {
  const result = runService("let's begin");
  console.log({ isMainThread });
}

run().catch(err => console.error(err));
```
```javascript
import { isMainThread, parentPort } from "node:worker_threads";

parentPort.on("message", message => {
  if (message === "exit") {
    parentPort.postMessage("sold!");
    parentPort.close();
  } else {
    parentPort.postMessage({ going: message });
  }
});

parentPort.postMessage({ start: workerData, isMainThread });
```
```
{
  "name": "worker",
  "version": "1.0.0",
  "description": "",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "author": "",
  "license": "ISC",
  "type": "module"
}
```

# **Log File**
Attached - had to rename to .log for github/markdown compatability 

[vscode-debugadapter-d0ecd913.log](https://github.com/microsoft/vscode-js-debug/files/10889821/vscode-debugadapter-d0ecd913.log)

# **VS Code Version: 1.76.0** 

# **Additional context**
The same is seemingly happening when running directly with ```node main.js``` as well. I tried to debug it using VSCode and then received the above situation/errors.

```
PS C:\src\worker> node main.js
C:\src\worker
{ isMainThread: true }
Worker stopped with exit code 1
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -10,6 +10,7 @@ This changelog records changes to stable releases since 1.50.2. "TBA" changes he
 - fix: unbound breakpoints in sourcemaps on Chrome 112 ([#1567](https://github.com/microsoft/vscode-js-debug/issues/1567))
 - fix: assorted bad source behaviors when reloading a page ([#1582](https://github.com/microsoft/vscode-js-debug/issues/1582))
 - fix: step over eval/new Function with sourcemaps not working ([#1556](https://github.com/microsoft/vscode-js-debug/issues/1556))
+- fix: 'break on caught exceptions' pausing on worker threads ([#1591](https://github.com/microsoft/vscode-js-debug/issues/1591))
 - chore: remove webpack, adopt esbuild
 
 ## v1.76 (February 2023)

--- a/src/targets/node/bootloader/filters.ts
+++ b/src/targets/node/bootloader/filters.ts
@@ -29,13 +29,12 @@ export const checkLeaseFile = (env: IBootloaderInfo) => {
 
 // Do not enable for Electron and other hybrid environments.
 export const checkNotElectron = () => {
-  try {
-    eval('window');
+  if (typeof window !== 'undefined') {
     bootloaderLogger.info(LogTag.RuntimeLaunch, 'Disabling in Electron (window is set)');
     return false;
-  } catch (e) {
-    return true;
   }
+
+  return true;
 };
 
 export const checkProcessFilter = (env: IBootloaderInfo) => {

